jars/src/org/toast/activegates/FlyThroughGate.kt

```diff
@@ -0,0 +1,71 @@
+package org.toast.activegates
+
+import com.fs.starfarer.api.Global
+import com.fs.starfarer.api.campaign.InteractionDialogAPI
+import com.fs.starfarer.api.campaign.rules.MemoryAPI
+import com.fs.starfarer.api.impl.campaign.rulecmd.BaseCommandPlugin
+import com.fs.starfarer.api.util.Misc
+import kotlin.math.roundToInt
+
+class FlyThroughGate : BaseCommandPlugin() {
+
+    override fun execute(ruleId: String?, dialog: InteractionDialogAPI?,
+                         params: List<Misc.Token>,
+                         memoryMap: Map<String, MemoryAPI>): Boolean {
+        if (dialog == null) return false
+
+        val textPanel = dialog.textPanel
+
+        // can only fly through activated gates
+        if (!dialog.interactionTarget.hasTag(GateCommandPlugin.TAG_GATE_ACTIVATED)) {
+            textPanel.addParagraph("Your fleet passes through the inactive gate...")
+            textPanel.addParagraph("and nothing happens.")
+            ShowGateDestinationOptions().execute(null, dialog, emptyList(), memoryMap)
+            return false
+        }
+
+        val systemIdChosenByPlayer = params[0].getStringWithTokenReplacement(ruleId, dialog, memoryMap)
+        if (systemIdChosenByPlayer == null || systemIdChosenByPlayer.isEmpty()) return false
+
+        val newSys = Global.getSector().getStarSystem(systemIdChosenByPlayer)
+
+        if (newSys == null) {
+            textPanel.addParagraph("Could not find $systemIdChosenByPlayer; aborting")
+            ShowGateDestinationOptions().execute(null, dialog, params, memoryMap)
+            return false
+        }
+
+        val playerFleet = Global.getSector().playerFleet
+
+        val cargo = playerFleet.cargo
+        val fuelcost = GateCommandPlugin.fuelCostPerLY * Misc.getDistanceLY(playerFleet.locationInHyperspace, newSys.location)
+        if (cargo.fuel >= fuelcost) {
+            cargo.removeFuel(fuelcost)
+        } else {
+            textPanel.addParagraph("Unfortunately, your fleet lacks the " + fuelcost.roundToInt() +
+                    " fuel necessary to use the gate.")
+            ShowGateDestinationOptions().execute(null, dialog, params, memoryMap)
+            return false
+        }
+
+        val oldSys = playerFleet.containingLocation
+        oldSys.removeEntity(playerFleet)
+        newSys.addEntity(playerFleet)
+        Global.getSector().currentLocation = newSys
+        val gates = newSys.getEntitiesWithTag(GateCommandPlugin.TAG_GATE)
+        val newVect = gates[0].location
+        playerFleet.setLocation(newVect.x, newVect.y)
+        playerFleet.clearAssignments()
+        playerFleet.setMoveDestination(playerFleet.location.x, playerFleet.location.y)
```

Comment:
what was the intent of this line?

Reply:
Whenever I'd jump through a gate, my fleet would start flying to the point in the new system that the gate we just jumped from had occupied. Like if gate A in system A is located at point A, and you jump to system B, my fleet would start flying to the location of point A but in system B.

I didn't spend much time making sure that I wasn't doing anything dumb, but adding this line of code fixed it.